src/safeposix/syscalls/net_calls.rs

```diff
@@ -3123,69 +3301,87 @@ impl Cage {
                         continue;
                     }
 
+                    // get the events to monitor
                     let events = value.events;
                     let mut structpoll = PollStruct {
                         fd: key,
                         events: 0,
                         revents: 0,
                     };
+                    // check for each supported event
                     if events & EPOLLIN as u32 > 0 {
                         structpoll.events |= POLLIN;
                     }
                     if events & EPOLLOUT as u32 > 0 {
                         structpoll.events |= POLLOUT;
                     }
-                    if events & EPOLLERR as u32 > 0 {
-                        structpoll.events |= POLLERR;
+                    if events & EPOLLPRI as u32 > 0 {
+                        structpoll.events |= POLLPRI;
                     }
+                    // now PollStruct is constructed, push it to the vector
                     poll_fds_vec.push(structpoll);
-                    num_events += 1;
                 }
 
                 for fd in rm_fds_vec.iter() {
                     epollfdobj.registered_fds.remove(fd);
                 } // remove closed fds
 
+                // call poll_syscall
                 let poll_fds_slice = &mut poll_fds_vec[..];
                 let pollret = Self::poll_syscall(&self, poll_fds_slice, timeout);
                 if pollret < 0 {
+                    // in case of error, return the error
                     return pollret;
                 }
+                // the counter is used for making sure the number of returned ready fds
+                // is smaller than or equal to maxevents
                 let mut count = 0;
-                let end_idx: usize = interface::rust_min(num_events, maxevents as usize);
-                for result in poll_fds_slice[..end_idx].iter() {
+                for result in poll_fds_slice.iter() {
+                    // transform the poll result into epoll result
+                    // poll_event is used for marking if the fd is ready
                     let mut poll_event = false;
                     let mut event = EpollEvent {
                         events: 0,
                         fd: epollfdobj.registered_fds.get(&result.fd).unwrap().fd,
                     };
+                    // check for POLLIN
```

Comment:
What's the difference between `events` and `revents`?

Reply:
`events` and `revents` are something used in `poll_syscall`. `events` is requested events, and `revents` is the return events that all results are stored here

Reply:
Can you please write this in the form of comments, so that once we come back to this syscall in future, we can understand it at first glance?